neo4j-jdbc-bolt/src/test/java/org/neo4j/jdbc/bolt/cache/BoltDriverCacheTest.java

```diff
@@ -20,6 +32,43 @@
 
     Function<BoltDriverCacheKey, Driver> builder = (params -> Mockito.mock(Driver.class));
 
+    @Test
+    public void cacheShouldBeThreadSafe() throws ExecutionException, InterruptedException {
+        List<URI> url = Arrays.asList(URI.create("bolt://localhost"));
+        Config.ConfigBuilder configBuilder = Config.builder();
+        AuthToken authToken = AuthTokens.basic("neo4j", "password");
+
+        BoltDriverCache cache = new BoltDriverCache(builder);
+        assertEquals(0, cache.getCache().size());
+
+        int numThreads = Runtime.getRuntime().availableProcessors();
+        ExecutorService executor = Executors.newWorkStealingPool();
+
+        AtomicBoolean running = new AtomicBoolean();
+        AtomicInteger overlaps = new AtomicInteger();
+
+        Collection<Callable<Driver>> getDriver = new ArrayList<>();
+        for (int t = 0; t < numThreads; ++t) {
+            getDriver.add(() -> {
+                if (!running.compareAndSet(false, true)) {
+                    overlaps.incrementAndGet();
+                }
+                Driver d = cache.getDriver(url, configBuilder.build(), authToken, new Properties());
+                running.compareAndSet(true, false);
+                return d;
+            });
+        }
+
+        Set<Driver> drivers = new IdentityHashSet<>();
+        for (Future<Driver> getDriverFuture : executor.invokeAll(getDriver)) {
+            drivers.add(getDriverFuture.get());
+        }
+
+        assertTrue(overlaps.get() > 0); // This might get flacky...
```

Comment:
```suggestion
        assertTrue(overlaps.get() > 0); // This might get flaky...
```

Reply:
unless there is a small number of threads (and if even then), overlaps size of 0 seems very unlikely or am I missing sth? what's the purpose of this assertion?

Reply:
doh! I initially missed the second CAS call!